akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -1342,3 +1342,47 @@ abstract class AbstractOutHandler extends OutHandler
  * (completing when upstream completes, failing when upstream fails, completing when downstream cancels).
  */
 abstract class AbstractInOutHandler extends InHandler with OutHandler
+
+/**
+ * This trait wraps callback for `GraphStage` stage instances and handle gracefully cases when stage is
+ * not yet initialized or already finished.
+ *
+ * While `GraphStage` has not initialized it adds all requests to list.
+ * As soon as `GraphStage` is started it stops collecting requests (pointing to real callback
+ * function) and run all the callbacks from the list
+ *
+ * Supposed to be used by GraphStages that share call back to outer world
+ */
+private[akka] trait CallbackWrapper[T] {
```

Comment:
Why not make it extend `Callback[T]` and rename `callback` to `invoke`? I do agree with the `private[akka]` scope at this point, and it should also get an `INTERNAL API` header in the comment.